manager/integration/tests/test_engine_upgrade.py

```diff
@@ -1197,18 +1203,67 @@ def test_engine_live_upgrade_while_replica_concurrent_rebuild(client, # NOQA
         assert replica.image == engine_upgrade_image
         assert replica.currentImage == engine_upgrade_image
 
-@pytest.mark.skip(reason="TODO")  # NOQA
-def test_engine_crash_during_live_upgrade():
+
+def test_engine_crash_during_live_upgrade(client, core_api, # NOQA
+                                          make_deployment_with_pvc, # NOQA
+                                          volume_name): # NOQA
     """
-    1. Create and attach a volume to a workload, then write data into the
+    1. Deploy an extra engine image.
+    2. Create and attach a volume to a workload, then write data into the
        volume.
-    2. Deploy an extra engine image.
     3. Send live upgrade request then immediately delete the related engine
        manager pod/engine process (The new replicas are not in active in this
        case).
     4. Verify the workload will be restarted and the volume will be reattached
        automatically.
-    5. Verify the upgrade is done during the reattachment.
+    5. Verify the upgrade is done.
        (It actually becomes offline upgrade.)
     6. Verify volume healthy and the data is correct.
     """
+    # Step 1
+    _, default_img_name, engine_upgrade_image, compatible_img, _ = \
+        prepare_auto_upgrade_engine_to_default_version(client)
+
+    # Step 2
+    host_id = get_self_host_id()
+    volume, pod_name, checksum, deployment = \
+        create_deployment_and_write_data(client, core_api,
+                                         make_deployment_with_pvc, # NOQA
+                                         volume_name, str(1 * Gi),
+                                         3,
+                                         DATA_SIZE_IN_MB_2,
+                                         host_id) # NOQA
+
+    create_snapshot(client, volume_name)
```

Comment:
There is a chance(about 1/5) to hit no data exist in the new deployment pod make this test case flaky. I found it's a timing issue between the end of step 2 and the beginning of step 3. I added `create_snapshot` before engine upgrade make sure data stored in all replicas.

Reply:
> hit no data exist in the new deployment pod make this test case flaky.

What does the error message look like? Where will it check the data existence between step 2 ~ 3? Is it possible to elaborate more? Thank you!

Reply:
Step 6 have a chance to hit `md5sum: can't open '/data/test': No such file or directory` or hit data md5sum not match.

It's timing issue, imagine you write 300MB data, while the volume actual size = 0, immediately you perfrom engine upgrade and crash. and hope the data was the same.

Using `create_snapshot` to force update the volume actual size to cope with, Sleep(20) can also work but obviously it's not a proper way.
```
    data_path = '/data/test'
    test_data_checksum = get_pod_data_md5sum(core_api,
                                             new_pod_name,
                                             data_path)
    assert test_data_checksum == checksum
```

Reply:
Is it about data didn't hit block device yet? Have you tried `conv=sync`, `oflag=sync` or separate `sync` command?

Reply:
Updated, remove `create_snapshot`, add `sync` command in `create_deployment_and_write_data` and verified on local.

Reply:
It would be better if `sync` can be wrapped in `write_pod_volume_random_data` function to make `write` and `sync` an atomic and cohesive operation, which also guarantees the `write` function did it job instead of just keeps data in memory.